PROTO_tests/tests/numeric_test.py

```diff
@@ -197,22 +205,159 @@ def test_cumprod(self, num_type):
         with pytest.raises(TypeError):
             ak.sin(np.array([range(0, 10)]).astype(num_type))
 
     @pytest.mark.parametrize("num_type", NO_BOOL)
     def test_cos(self, num_type):
-        na = np.linspace(1, 10, 10).astype(num_type)
+        na = NP_TRIG_ARRAYS[num_type]
         pda = ak.array(na, dtype=num_type)
 
-        assert np.allclose(np.cos(na), ak.cos(pda).to_ndarray())
+        assert np.allclose(np.cos(na), ak.cos(pda).to_ndarray(), equal_nan=True)
         with pytest.raises(TypeError):
             ak.cos(np.array([range(0, 10)]).astype(num_type))
 
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_tan(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.tan(na), ak.tan(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.tan(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arcsin(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arcsin(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arccos(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arccos(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arctan(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arctan(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    @pytest.mark.parametrize("denom_type", NO_BOOL)
+    def test_arctan2(self, num_type, denom_type):
+        na_num = np.random.permutation(NP_TRIG_ARRAYS[num_type])
+        na_denom = np.random.permutation(NP_TRIG_ARRAYS[denom_type])
+
+        pda_num = ak.array(na_num, dtype=num_type)
+        pda_denom = ak.array(na_denom, dtype=denom_type)
+
+        num = np.array([1]).astype(num_type)
+        denom = np.array([5]).astype(denom_type)
+
+        assert np.allclose(
+            np.arctan2(na_num, na_denom), ak.arctan2(pda_num, pda_denom).to_ndarray(), equal_nan=True
+        )
+        assert np.allclose(
+            np.arctan2(num[0], na_denom), ak.arctan2(num[0], pda_denom).to_ndarray(), equal_nan=True
+        )
+        assert np.allclose(
+            np.arctan2(na_num, denom[0]), ak.arctan2(pda_num, denom[0]).to_ndarray(), equal_nan=True
+        )
```

Comment:
how about we update to something like this so we aren't testing the same scalars everytime
```suggestion
        num_scal = pda_num[0]
        denom_scal = pda_denom[0]

        assert np.allclose(
            np.arctan2(na_num, na_denom), ak.arctan2(pda_num, pda_denom).to_ndarray(), equal_nan=True
        )
        assert np.allclose(
            np.arctan2(num_scal, na_denom), ak.arctan2(num_scal, pda_denom).to_ndarray(), equal_nan=True
        )
        assert np.allclose(
            np.arctan2(na_num, denom_scal), ak.arctan2(pda_num, denom_scal).to_ndarray(), equal_nan=True
        )
```